compiler/modules/front.syntax/syntaxNode.bal

```diff
@@ -0,0 +1,830 @@
+                                              members,
+                                              { token: "]" });
+}
+
+function syntaxNodeFromMappingConstructorExpr(MappingConstructorExpr expr) returns NonTerminalSyntaxNode{
+    SyntaxNode[] fields = joinSyntaxNodesWithSeperator((from Field f in expr.fields select syntaxNodeFromField(f)), { token: ",", outputFlag: NEWLINE_AFTER });
+    return nonTerminalSyntaxNode(expr, false, { token: "{", pos: expr.startPos },
+                                              // JBUG: can't use the query exprssion directly
+                                              fields,
+                                              { token: "}" });
+}
+
+function syntaxNodeFromMemberAccessExpr(MemberAccessExpr expr, boolean wrap) returns NonTerminalSyntaxNode {
+    return nonTerminalSyntaxNode(expr, wrap, syntaxNodeFromExpr(expr.container, true),
+                                             { token: "[", outputFlag: CLING_PREV },
+                                             syntaxNodeFromExpr(expr.index, false),
+                                             { token: "]"});
+}
+
+function syntaxNodeFromFieldAccessExpr(FieldAccessExpr expr, boolean wrap) returns NonTerminalSyntaxNode {
+    return nonTerminalSyntaxNode(expr, wrap, syntaxNodeFromExpr(expr.container, true),
+                                             { token: ".", pos: expr.opPos },
+                                             { name: expr.fieldName, pos: () });
+}
+
+function syntaxNodeFromTypeCastExpr(TypeCastExpr expr, boolean wrap) returns NonTerminalSyntaxNode {
+    return nonTerminalSyntaxNode(expr, wrap, { token: "<", outputFlag: CLING_NEXT },
+                                             syntaxNodeFromTypeDesc(expr.td, false),
+                                             { token: ">", outputFlag: CLING },
+                                             syntaxNodeFromExpr(expr.operand, true));
+}
+
+function syntaxNodeFromVarRefExpr(VarRefExpr expr) returns NonTerminalSyntaxNode {
+    string? prefix = expr.prefix;
+    Position? identifierPos = prefix == () ? expr.qNamePos : ();
+    return nonTerminalSyntaxNode(expr, false, prefix != () ? [{ name: prefix, pos: expr.qNamePos }, { token: ":", outputFlag: CLING }] : (),
+                                              { name: expr.name, pos: identifierPos });
+}
+
+function syntaxNodeFromCheckingExpr(CheckingExpr expr) returns NonTerminalSyntaxNode {
+    return nonTerminalSyntaxNode(expr, false, { token: expr.checkingKeyword, pos: expr.kwPos },
+                                              syntaxNodeFromExpr(expr.operand, false));
+}
+
+function syntaxNodeFromUnaryExpr(UnaryExpr expr, boolean wrap) returns NonTerminalSyntaxNode {
+    return nonTerminalSyntaxNode(expr, wrap, { token: expr.op, pos: expr.opPos, outputFlag: CLING_NEXT },
+                                             syntaxNodeFromExpr(expr.operand, true));
+}
+
+type TerminalExpr LiteralExpr|NumericLiteralExpr;
```

Comment:
I don't like `TerminalExpr`.  We should rename LiteralExpr to something else (e.g. NonNumericLiteralExpr) and then define LiteralExpr to be NumericLiteralExpr|NonNumericLiteralExpr.

But NonNumericLiteralExpr is a bit long.  The point is that it is something where we don't need to the contextually expected to resolve it. Let's call it FixedLiteralExpr (unless you can think of something better).

Reply:
fixed